script.js

```diff
@@ -1,4 +1,205 @@
+      chooseNumberArray.push(doubleDigitNumber);
+
+      gameMode = diceRollGameModePlayer2;
+      return `Hi Player 1, you have chosen ${doubleDigitNumber}. Now it is Player 2's turn to play.`;
+    }
+    if (input == 2) {
+      doubleDigitNumber = parseInt(`${diceRollArray1[1]}${diceRollArray1[0]}`);
+      chooseNumberArray.push(doubleDigitNumber);
+
+      gameMode = diceRollGameModePlayer2;
+
+      return `Hi Player 1, you have chosen ${doubleDigitNumber}. Now it is Player 2's turn to play.`;
+    }
+    if (input != 1 || input != 2) {
+      return `Hi Player 1, you have not typed in 1 or 2, please type in 1 or 2 <br> <br>
+      You have rolled Dice 1 as ${diceRollArray1[0]} and Dice 2 as ${diceRollArray1[1]}. <br> <br>`;
+    }
+  }
+
+  if (gameMode == diceRollGameModePlayer2) {
+    // define variable for PLayer 2's roll dice array
+    var myOutputValue2 = rollAndStoreDicePlayer2();
+
+    //change game mode to dice order game mode
+    gameMode = diceOrderGameModePlayer2;
+
+    return myOutputValue2;
+  }
+
+  // roll dice and set order for Player 2
+  if (gameMode == diceOrderGameModePlayer2) {
+    if (input == 1) {
+      doubleDigitNumber = parseInt(`${diceRollArray2[0]}${diceRollArray2[1]}`);
+      chooseNumberArray.push(doubleDigitNumber);
+
+      console.log(gameMode);
+
+      // move to compare score game mode
+      gameMode = compareScoreGameMode;
+      return `Hi Player 2, you have chosen ${doubleDigitNumber}.`;
+    }
+    if (input == 2) {
+      doubleDigitNumber = parseInt(`${diceRollArray2[1]}${diceRollArray2[0]}`);
+      chooseNumberArray.push(doubleDigitNumber);
+      console.log(chooseNumberArray);
+
+      console.log(gameMode);
+
+      // move to compare score game mode
+      gameMode = compareScoreGameMode;
+      return `Hi Player 2, you have chosen ${doubleDigitNumber}.`;
+    }
+    if (input != 1 || input != 2) {
+      return `Hi Player 2, you have not typed in 1 or 2, please type in 1 or 2 <br> <br>
+      You have rolled Dice 1 as ${diceRollArray2[0]} and Dice 2 as ${diceRollArray2[1]}. <br> <br>`;
+    }
+  }
+
+  // Determine winner
+  if (gameMode == compareScoreGameMode) {
+    // player 1 wins condition
+    if (chooseNumberArray[0] > chooseNumberArray[1]) {
+      // reset to player 1 game mode
+      gameMode = diceRollGameModePlayer1;
+      // show winning message
+      return `Hi Player 1, you have won! Your number is ${chooseNumberArray[0]}, Player 2's number is ${chooseNumberArray[1]}.`;
+    }
+    // player 2 wins condition
+    if (chooseNumberArray[1] > chooseNumberArray[0]) {
+      // reset to player 1 game mode
+      gameMode = diceRollGameModePlayer1;
+      // show winning message
+      return `Hi Player 2, you have won! Your number is ${chooseNumberArray[1]}, Player 1's number is ${chooseNumberArray[0]}.`;
+    }
+    // draw condition
+    if (chooseNumberArray[0] == chooseNumberArray[1]) {
```

Comment:
You can actually omit the if condition here (line 160). Because if the previous two if conditions are not met, this is the only scenario left - therefore this is the default scenario. If you remove this if condition, your code will work perfectly too + neater!